test/pylib/scylla_cluster.py

```diff
@@ -0,0 +1,792 @@
+        self.sock_path: str = f"{self.harness_dir}/api"
+        self.app = aiohttp.web.Application()
+        self._setup_routes()
+        self.runner = aiohttp.web.AppRunner(self.app)
+        self.site: Optional[aiohttp.web.UnixSite] = None
+
+    async def start(self) -> None:
+        """Start Harness, setup API, start first cluster"""
+        await self._before_test(self.test_name)  # Call with pytest name
+        await self.runner.setup()
+        self.site = aiohttp.web.UnixSite(self.runner, path=self.sock_path)
+        await self.site.start()
+        self.is_running = True
+
+    async def _before_test(self, test_name: str) -> None:
+        if self.cluster is not None and self.cluster.is_dirty:
+            await self.cluster.stop()
+            self.cluster = None
+        if self.cluster is None:
+            # TODO: if cluster startup takes too long and HTTP timeouts, return immediately
+            # (i.e. large clusters with many servers)
+            await self._get_cluster()
+        assert self.cluster is not None, "Missing cluster before test"
+        logging.info("Leasing Scylla cluster %s for test %s", self.cluster, test_name)
+        self.cluster.before_test(self.test_name)
+        self.is_before_test_ok = True
+
+    async def stop(self) -> None:
+        """Stop Harness, cycle last cluster if not dirty and present"""
+        if self.site is not None:
+            await self.site.stop()
+            self.site = None
+        if self.cluster is not None:
+            self.cluster.after_test(self.test_name)
+            if self.cluster.is_dirty:
+                await self.cluster.stop()
+                self.cluster = None
+            else:
+                logging.info("Returning Scylla cluster %s", self.cluster)
+                await self.clusters.put(self.cluster)  # Return
+                self.cluster = None
+        if os.path.exists(self.harness_dir):
+            shutil.rmtree(self.harness_dir)
+        self.is_after_test_ok = False
+        self.is_running = False
+
+    async def _get_cluster(self) -> None:
+        assert self.cluster is None, "Previous cluster should be stopped"
+        self.cluster = await self.clusters.get()
+        logging.info("Getting new Scylla cluster %s", self.cluster)
+
+    def _setup_routes(self) -> None:
+        self.app.router.add_get('/up', self._harness_up)
+        self.app.router.add_get('/cluster/up', self._cluster_up)
+        self.app.router.add_get('/cluster/is-dirty', self._is_dirty)
+        self.app.router.add_get('/cluster/replicas', self._cluster_replicas)
+        self.app.router.add_get('/cluster/servers', self._cluster_servers)
+        self.app.router.add_get('/cluster/before_test/{test_name}', self._before_test_req)
+        self.app.router.add_get('/cluster/after_test/{test_name}', self._after_test)
+        self.app.router.add_get('/cluster/mark-dirty', self._mark_dirty)
```

Comment:
I think we agreed to follow _ vs - rules consistently, and settled on underscores. Please rename is-dirty -> is_dirty, mark-dirty -> mark_dirty. Thanks.

Reply:
This was a mistake. URL should've been `/before-test`  `/after-test`